pkg/clustermesh/mcsapi/service_controller.go

```diff
@@ -166,23 +160,25 @@ func (r *mcsAPIServiceReconciler) getBaseDerivedService(
 	return &svc, true, nil
 }
 
-func (r *mcsAPIServiceReconciler) getLocalService(ctx context.Context, req ctrl.Request) (*corev1.Service, error) {
-	var svc corev1.Service
-	if err := r.Client.Get(ctx, req.NamespacedName, &svc); err != nil {
+// getLocalServiceIfExported returns the service that we are currently exporting from. This
+// means that that the Service is only returned if a ServiceExport is also created
+func (r *mcsAPIServiceReconciler) getLocalServiceIfExported(ctx context.Context, req ctrl.Request) (*corev1.Service, error) {
+	var svcExport mcsapiv1alpha1.ServiceExport
+	if err := r.Client.Get(ctx, req.NamespacedName, &svcExport); err != nil {
+		if k8sApiErrors.IsNotFound(err) {
+			return nil, nil
+		}
 		return nil, err
```

Comment:
```suggestion
		return nil, client.IgnoreNotFound(err)
```

Dropping the if above. Same below.